src/support/archive.cpp

```diff
@@ -133,7 +133,7 @@ Archive::Child Archive::Child::getNext(bool& error) const {
   // Members are aligned to even byte boundaries.
   if (toSkip & 1) ++toSkip;
   const uint8_t* nextLoc = data + toSkip;
-  if (nextLoc >= (uint8_t*)&*parent->data.end()) {  // End of the archive.
+  if (nextLoc >= (const uint8_t*)parent->data.data() + parent->data.size()) {  // End of the archive.
```

Comment:
Oh, actually no, it's not; its the start of the data for the archive member represented by `this`. So this could be `if ((data - (const uint8_t)parent->data.data()) + toSkip >= parent->data.size())` which has the advantage that we never even create a pointer that points outside `parent->data()`.